Utilities/CreateNewExample.py.in

```diff
@@ -1,529 +0,0 @@
-#!/usr/bin/env python
-
-""" Scripts to create a new example
-
-./CreateNewExample.py /path/to/source/of/ITK
-"""
-
-import os
-import sys
-import argparse
-import re
-import fnmatch
-import collections
-
-try:
-    # Python 3
-    from io import StringIO
-except ImportError:
-    # Python 2
-    from cStringIO import StringIO
-
-get_input = input
-if sys.version_info[:2] <= (2, 7):
-    get_input = raw_input
-
-
-def write_all_itk_headers(itk_source_dir):
-    output = StringIO()
-    for root, dirs, files in os.walk(os.path.join(itk_source_dir, 'Modules')):
-        for f in fnmatch.filter(files, 'itk*.h'):
-            output.write(f + u'\n')
-    return output
-
-
-def words(text):
-    temp = re.findall('itk[A-Z][a-zA-Z_0-9]+', text)
-    for i in range(len(temp)):
-        var = temp[i]
-        temp[i] = var[3:]
-    return temp
-
-
-def train(features):
-    model = collections.defaultdict(lambda: 1)
-    for f in features:
-        model[f] += 1
-    return model
-
-
-def edits1(word):
-    splits = [(word[:i], word[i:]) for i in range(len(word) + 1)]
-    deletes = [a + b[1:] for a, b in splits if b]
-    transposes = [a + b[1] + b[0] + b[2:] for a, b in splits if len(b) > 1]
-    replaces = [a + c + b[1:] for a, b in splits for c in alphabet if b]
-    inserts = [a + c + b for a, b in splits for c in alphabet]
-    lowercases = [a + b[0].lower() + b[1:] for a, b in splits if b]
-    uppercases = [a + b[0].upper() + b[1:] for a, b in splits if b]
-    all_lowercase = [word[:].lower()]
-    one_uppercase = [
-        a[:].lower() + b[0].upper() + b[1:].lower() for a, b in splits if b]
-
-    return set(
-        deletes + transposes + replaces + inserts + lowercases + uppercases +
-        all_lowercase + one_uppercase)
-
-
-def known_edits2(word):
-    return set(e2 for e1 in edits1(word) for e2 in edits1(e1) if e2 in NWORDS)
-
-
-def known(words):
-    return set(w for w in words if w in NWORDS)
-
-
-def itkcorrect(word):
-    candidates = (
-        known([word]) or known(edits1(word)) or known_edits2(word) or [word])
-    return max(candidates, key=NWORDS.get)
-
-
-def space_out_camel_case(s):
-    return re.sub('((?=[A-Z][a-z])|(?<=[a-z])(?=[A-Z]))', ' ', s)
-
-
-def get_group_and_module_from_class_name(itk_dir, class_name):
```

Comment:
Can we re-use this to automatically get the group and module name?

Reply:
Maybe with the hooks?

 https://cookiecutter.readthedocs.io/en/latest/advanced/hooks.html

Reply:
A first attempt done in e3143ac.